### PR TITLE
Resolves #162 adds camel-box support

### LINKED ISSUE #162: support camel-box

### PR DESCRIPTION
**Pull request by johnpoth**
Thanks !

### CODE REVIEW

File: extensions/box/deployment/src/main/java/org/apache/camel/quarkus/component/box/deployment/BoxProcessor.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    @BuildStep
+    AdditionalApplicationArchiveMarkerBuildItem boxArchiveMarker() {
+        return new AdditionalApplicationArchiveMarkerBuildItem("com/box/sdk");
+    }
+
+    @BuildStep
+    void configure(BuildProducer<SystemPropertyBuildItem> systemProperties,
+            BuildProducer<RuntimeInitializedClassBuildItem> runtimeClasses) {
+        systemProperties.produce(new SystemPropertyBuildItem("io.netty.noUnsafe", "true"));
```
Comment (oscerd) [inline extensions/box/deployment/src/main/java/org/apache/camel/quarkus/component/box/deployment/BoxProcessor.java]: Can 'io.netty.noUnsafe' be a variable?

Comment (ppalaga) [reply extensions/box/deployment/src/main/java/org/apache/camel/quarkus/component/box/deployment/BoxProcessor.java]: I do not see that box would depend on netty. Is the property really necessary?

File: integration-tests/box/README.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+Then set the following environment variables:
+
+[source,shell]
+----
+export BOX_USER_NAME=my-user-name
+export BOX_USER_PASSWORD=my-user-password
+export BOX_CLIENT_ID=my-client-id
+export BOX_CLIENT_SECRET=my-client_secret
+----
```
Comment (ppalaga) [inline integration-tests/box/README.adoc]: I appreciate the readme!

File: integration-tests/box/pom.xml [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                                    <graalvmHome>${graalvmHome}</graalvmHome>
+                                    <enableJni>true</enableJni>
+                                    <enableAllSecurityServices>true</enableAllSecurityServices>
+                                    <disableReports>true</disableReports>
+                                    <additionalBuildArgs>-H:+TraceClassInitialization</additionalBuildArgs>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
```
Comment (ppalaga) [inline integration-tests/box/pom.xml]: Could you please change this to the new style introduced in https://github.com/apache/camel-quarkus/pull/720/files#diff-13fed0010c65738aa32aa3e15e89171dR92-R114 ?

File: integration-tests/box/src/test/java/org/apache/camel/quarkus/component/box/it/BoxIT.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.box.it;
+
+import io.quarkus.test.junit.NativeImageTest;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+@EnabledIfEnvironmentVariable(named = "BOX_USER_NAME", matches = ".*")
```
Comment (ppalaga) [inline integration-tests/box/src/test/java/org/apache/camel/quarkus/component/box/it/BoxIT.java]: Hm... this seems to match an empty `BOX_USER_NAME`. I am not sure defining an empty env var is possible though. Anyway, something like `.+` would perhaps be better, WDYT?

File: integration-tests/pom.xml [hunk 5]
```diff
@@ -217,6 +217,7 @@
         <module>validator</module>
         <module>xslt</module>
         <module>zipfile</module>
+        <module>box</module>
```
Comment (ppalaga) [inline integration-tests/pom.xml]: Could you please sort the modules by running `mvn process-resources -Pformat` from the top level folder?

**Comment by johnpoth** [comment]
@oscerd @ppalaga addressed comments, thanks for reviewing !

**Pull request merged**